### PR TITLE
EM-888: Modal to export project details to CSV.

### PR DESCRIPTION
**Pull request by NickPhura**
Commit 1: code changes
Commit 2: just formatting (no need to review)
Commit 3: code review changes

Added an option to the admin drop down called 'Export Data'.
It spawns a little modal with a 'Download' button, that fetches all of the records from the Projects collection, and parses the top level fields into a csv file for download.

### CODE REVIEW

File: modules/core/client/directives/core.client.directives.js [hunk 1]
```diff
@@ -713,4 +713,59 @@ angular.module('core')
...
+              var s = this;
+              s.isLoading = false;
+              $scope.url = null;
+              s.exportProjects = function () {
+                s.isLoading = true;
+                ProjectModel.exportProjects().then(function (csvData) {
+                  var blob = new Blob([csvData], {
+                    type: 'text/csv'
+                  });
+                  $scope.download = 'epic_project_data_' + $filter('date')(new Date(), 'yyyy-MM-dd') + '.csv';
```
Comment (ActionAnalytics) [inline modules/core/client/directives/core.client.directives.js]: Is there a chance an Admin would check this twice in one day?  First time to see who needs updating, second to confirm the changes are how they wish them to be?

Comment (NickPhura) [reply modules/core/client/directives/core.client.directives.js]: Added hours/mins/secs to the filename

File: modules/projects/server/controllers/project.controller.js [hunk 2]
```diff
@@ -1365,5 +1367,120 @@ module.exports = DBModel.extend ({
...
+            _id: 0,
+            name: 1
+          }
+        }
+      ])
+      .sort('name')
+      .lean()
+      .exec()
+      .then(function (data) {
+        var jsonData = JSON.parse(JSON.stringify(data));
```
[the remark this answers is not in the document]
Comment (NickPhura) [reply modules/projects/server/controllers/project.controller.js]: I tried using this, and it didnt fix a couple of the weird characters.  So im thinking maybe its a weird copy-paste and those characters are actually in the database.

[the remark this answers is not in the document]
Comment (ActionAnalytics) [reply modules/projects/server/controllers/project.controller.js]: Fair enough, thanks for trying.

**Pull request merged**